### PR TITLE
[xpu][fix] Fix LinearCrossEntropyLoss-related UT accuracy on XPU

### PR DESCRIPTION
**Pull request by guangyey**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #181878
* __->__ #181859
* #181858

# Motivation
https://github.com/pytorch/pytorch/pull/181573 add `nn.LinearCrossEntropyLoss` but it only update the tolerance for CUDA, resulting in some cases failing on XPU.
This PR would update tolerance for XPU to pass them.

### CODE REVIEW

**Review by pearu — APPROVED** [review]
LGTM! Thanks, @guangyey!

**Comment by slayton58** [comment]
@pearu @guangyey Out of interest, why are the bounds so high for this test? Just numerically different to unfused?

**Comment by slayton58** [comment]
@guangyey test fails look... weird. Don't __think__ they're related, but they're for different reasons, and, well, weird :) Might be worth a rebase to main, see if it gets rid of any

**Comment by pearu** [comment]
> @pearu @guangyey Out of interest, why are the bounds so high for this test? Just numerically different to unfused?

I looked into the `test_cpu_gpu_parity` case and and let me just copy the explanation from a Claude session here as it is pretty accurate:
"""
  ...

  The actual root cause: matmul algorithm divergence, not softmax                                                                                                                       
                                                                                                                                                                                        
  Looking at AccumulateType.h, both CPU and CUDA use float32 as the accumulation type for float16/bfloat16 (lines 128–131 for CUDA, lines 149–150 for CPU). So the CPU softmax kernel   
  (SoftMax.cpp:219) and the CUDA softmax kernel both accumulate in float32 — that's symmetric.                                                                                          
                                                                                                                                                                                        
  The divergence comes from the matmul step (linear(input, linear_weight)):                                                                                                             
  
  - CPU matmul (float16/bfloat16): uses CPU BLAS or a scalar loop; the summation order and intermediate rounding follow the CPU's sequential or vectorized (AVX/AMX) algorithm          
  - GPU matmul: uses CUDA tensor cores, which tile the computation into warps/blocks with a fundamentally different summation tree — even with float32 accumulation internally, the
  result stored back in float16/bfloat16 is quantized differently                                                                                                                       
                  
  The logits (output of the matmul) are stored in float16/bfloat16. Once those differ — even by 1 ULP — the exp(logit_i - max) in softmax amplifies the difference nonlinearly, because exp has slope equal to its own value (large logits → large amplification). The subsequent log(sum_exp) has no way to recover that error.
                                                                                                                                                                                
  This is entirely a cross-entropy issue as the comment at line 4564 says — the LinearCrossEntropyLoss reference path just calls linear → cross_entropy, and the precision gap is inherited directly from the matmul divergence, amplified by the exp/log pipeline.
                                                                                                                                                                                        
  Why bfloat16 is expectedFailure rather than high-tolerance: bfloat16's 7-bit mantissa (vs float16's 10-bit) means the logit quantization error after matmul is larger, and the exp amplification produces divergences that no fixed tolerance can cover reliably across all test inputs.
"""

Similarly, in the case of `test_save_load`, the logits errors amplify in exp/log calls, although, the details of the root cause are different from the `test_cpu_gpu_parity` case (not shown here).

Btw, in https://github.com/pytorch/pytorch/pull/181574 we'll introduce `acc_dtype` to improve the accuracy with float16/bfloat16 dtypes.

**Review by slayton58 — APPROVED** [review]
This looks ok to me - stamping, but we should get the CI as close to green as humanly possible before merge (noting that CI feels.. temperamental today)

**Review by jansel — APPROVED** [review]

**Comment by guangyey** [comment]
These flaky failures are unrelated.

**Comment by guangyey** [comment]
Land this PR to fix these failures in XPU CI.
@pytorchbot merge

**Comment by guangyey** [comment]
I will investigate the remaining failures.
